src/modules/hip/kernel/image_sum.hpp

```diff
@@ -0,0 +1,481 @@
+                                                      uint xBufferLength,
+                                                      float *dstPtr)
+{
+    int id_x = hipThreadIdx_x * 8;
+    int id_z = hipBlockIdx_z;
+
+    __shared__ float partialRSumLDS[1024];                            // 2048 floats of src reduced to 1024 in a 1024 x 1 thread block
+    __shared__ float partialGSumLDS[1024];
+    __shared__ float partialBSumLDS[1024];
+    partialRSumLDS[hipThreadIdx_x] = 0.0f;                           // initialization of LDS to 0 using all 256 x 1 threads
+    partialGSumLDS[hipThreadIdx_x] = 0.0f;
+    partialBSumLDS[hipThreadIdx_x] = 0.0f;
+
+    if (id_x >= xBufferLength)
+    {
+        return;
+    }
+
+    int xAlignedLength = xBufferLength & ~7;                        // alignedLength for vectorized global loads
+    int xDiff = xBufferLength - xAlignedLength;               // difference between bufferLength and alignedLength
+    uint srcIdx = ((id_z * xBufferLength) + id_x) * 3;
+
+    d_float24 src_f24;
+    rpp_hip_load24_pkd3_and_unpack_to_float24_pln3(srcPtr + srcIdx, &src_f24);           // load 24 pixels to local mmemory
+    if (id_x + 8 > xBufferLength)                                                        // local memory reset of invalid values (from the vectorized global load) to 0.0f
+    {
+        for(int i = xDiff; i < 8; i++)
+        {
+            src_f24.f8[0].f1[i] = 0.0f;
+            src_f24.f8[1].f1[i] = 0.0f;
+            src_f24.f8[2].f1[i] = 0.0f;
+        }
+    }
+    src_f24.f8[0].f4[0] += src_f24.f8[0].f4[1];                                           // perform small work of vectorized float4 addition
+    src_f24.f8[1].f4[0] += src_f24.f8[1].f4[1];
+    src_f24.f8[2].f4[0] += src_f24.f8[2].f4[1];
+    partialRSumLDS[hipThreadIdx_x] = (src_f24.f8[0].f1[0] +
+                                      src_f24.f8[0].f1[1] +
+                                      src_f24.f8[0].f1[2] +
+                                      src_f24.f8[0].f1[3]);                   // perform small work of reducing R float4s to float using 16 x 16 threads and store in LDS
+    partialGSumLDS[hipThreadIdx_x] = (src_f24.f8[1].f1[0] +
+                                      src_f24.f8[1].f1[1] +
+                                      src_f24.f8[1].f1[2] +
+                                      src_f24.f8[1].f1[3]);                   // perform small work of reducing G float4s to float using 16 x 16 threads and store in LDS
+    partialBSumLDS[hipThreadIdx_x] = (src_f24.f8[2].f1[0] +
+                                      src_f24.f8[2].f1[1] +
+                                      src_f24.f8[2].f1[2] +
+                                      src_f24.f8[2].f1[3]);                  // perform small work of reducing B float4s to float using 16 x 16 threads and store in LDS
+
+    __syncthreads();                                                // syncthreads after LDS load
```

Comment:
Pls align all comments vertically with decent tabbing. Some lines that are too long can have non aligned comments at the end of line. Pls review this for host and hip code